Tests/GRPCHTTP2TransportTests/HTTP2TransportNIOPosixTests.swift

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import GRPCHTTP2Core
+import GRPCHTTP2TransportNIOPosix
+import XCTest
+
+@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+final class HTTP2TransportNIOPosixTests: XCTestCase {
+  func testGetListeningAddress_IPv4() async throws {
+    let transport = GRPCHTTP2Core.HTTP2ServerTransport.Posix(
+      address: .ipv4(host: "0.0.0.0", port: 0)
+    )
+
+    try await withThrowingDiscardingTaskGroup { group in
+      group.addTask {
+        try await transport.listen { _ in }
+      }
+
+      group.addTask {
+        let address = try await transport.listeningAddress
+        XCTAssertNotNil(address.ipv4)
+        XCTAssertNotEqual(address.ipv4!.port, 0)
```

Comment:
nit: prefer using `XCTUnwrap`

```suggestion
        let ipv4 = try XCTUnwrap(address?.ipv4)
        XCTAssertNotEqual(ipv4.port, 0)
```